### ISSUE TITLE
[BUG] Full-Screen issue

### ISSUE THREAD

**Issue by aseef17** [jhomlala/betterplayer#283]
**Describe the bug**
There seems to be some bug while trying to change the mode from full-screen to normal. This happens only the first time when the betterplayer is used. 

For example:- If a user enters a screen with the betterplayer and tries clicking on the full-screen button, it works, but when the user clicks on the full-screen exit icon, the betterplayer stays in full-screen mode. Clicking on it again takes the user back to portrait mode. This would repeat if the user leaves and enters the screen again. Please find the attached video

**To Reproduce**
1. Click on full-screen icon. (Works as expected)
2. Click on exit full-Screen icon. (Nothing happens). Click on it again and it starts working as expected

**Expected behavior**
The full-screen toggle should work fine without it glitching for the first time

**Video**
https://user-images.githubusercontent.com/26878414/107223225-eb67df80-6a3b-11eb-8b9a-ec82a860bb76.mov

**Flutter doctor**
<details>
<summary>Click to expand</summary>
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.2 20D64 darwin-x64, locale
    en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.53.0)
[✓] Connected device (1 available)

• No issues found!
</details>

**Better Player version**
- Version: 0.0.52

**Smartphone (please complete the following information):**
 - Device: iPhone SE2
 - OS: iOS 14.4

**Additional context**
I tried to log the `betterPlayerEventType` in `addEventsListener`, it does display `openFullscreen` the first time but shows `hideFullscreen` only when clicked on twice

**Relevant Code**
```
  /// Variables
  BetterPlayerController _betterPlayerController;

  Orientation _orientationBeforeFullScreen;
  final List<DeviceOrientation> _normalOrientations = [
    DeviceOrientation.portraitUp,
  ];
  final List<DeviceOrientation> _portraitOrientation = [
    DeviceOrientation.portraitUp,
  ];
  final List<DeviceOrientation> _landscapeOrientation = [
    DeviceOrientation.landscapeLeft,
    DeviceOrientation.landscapeRight
  ];
  
      /// initState Code
      final BetterPlayerDataSource dataSource = BetterPlayerDataSource(
      BetterPlayerDataSourceType.network,
      widget.url.trim().toString(),
    );
    _betterPlayerController = BetterPlayerController(betterPlayerConfiguration);
    _betterPlayerController.setupDataSource(dataSource);

    _betterPlayerController.addEventsListener((event) {
      debugPrint('${event.betterPlayerEventType}');

      ///On fullscreen event check current Orientation
      if (event.betterPlayerEventType == BetterPlayerEventType.openFullscreen) {
        _orientationBeforeFullScreen = MediaQuery.of(context).orientation;
      }

      ///On fullscreen finish set DeviceOrientation based on saved Orientation
      if (event.betterPlayerEventType == BetterPlayerEventType.hideFullscreen) {
        if (_orientationBeforeFullScreen == Orientation.portrait) {
          SystemChrome.setPreferredOrientations(_portraitOrientation);
        }
        if (_orientationBeforeFullScreen == Orientation.landscape) {
          SystemChrome.setPreferredOrientations(_landscapeOrientation);
        }

        ///Once page correctly render in given Orientation, add normal
        ///orientations, so user can use both portrait and landscape.
        Future.delayed(const Duration(milliseconds: 500), () {
          SystemChrome.setPreferredOrientations(_normalOrientations);
        });
      }
    });
    
   
    /// Widget Code
    AspectRatio(
          aspectRatio: 16 / 9,
          child: BetterPlayer(controller: _betterPlayerController),
    ),
```

**Comment by isanimu** [comment]
@aseef17 I don't know if this will help you or not. But I think you need to await.

```dart
_betterPlayerController.addEventsListener((event) async {
  if (event.betterPlayerEventType == BetterPlayerEventType.hideFullscreen) {
    if (_orientationBeforeFullScreen == Orientation.portrait) {
      await SystemChrome.setPreferredOrientations(_portraitOrientation);
    }
    if (_orientationBeforeFullScreen == Orientation.landscape) {
      await SystemChrome.setPreferredOrientations(_landscapeOrientation);
    }
  }
}
```
Since this also happened to me before. And also , for every BetterPlayerController api that is async you need to await for it like for example pause & play.

**Comment by jorelkimcruz** [comment]
@aseef17 I think you just need to set 
 deviceOrientationsAfterFullScreen: [DeviceOrientation.portraitUp],

**Comment by jhomlala** [comment]
Fixed in 0.0.53
